custom/enikshay/tests/test_adherence_updates.py

```diff
@@ -0,0 +1,363 @@
+
+        return EpisodeUpdate(episode, self.case_updater)
+
+    def test_adherence_schedule_date_start_late(self):
+        #   Sample test case
+        #   [
+        #       (
+        #           purge_date,
+        #           (adherence_schedule_date_start, adherence_schedule_id),
+        #           [
+        #               (adherence_date, adherence_value),
+        #               (adherence_date, adherence_value),
+        #               ...
+        #           ],
+        #           {
+        #               'aggregated_score_date_calculated': value
+        #               'expected_doses_taken': value
+        #               'aggregated_score_count_taken': value
+        #           }
+        #       ),
+        #       ...
+        #   ]
+
+        self.assert_update(
+            (
+                datetime(2016, 1, 15),
+                (datetime(2016, 1, 17), 'schedule1'),
+                []
+            ),
+            {
+                'aggregated_score_date_calculated': date(2016, 1, 16),
+                'expected_doses_taken': 0,
+                'aggregated_score_count_taken': 0,
+                # 1 day before should be adherence_schedule_date_start,
+                'adherence_latest_date_recorded': date(2016, 1, 16),
+                'adherence_total_doses_taken': 0
+            }
+        )
+
+    def test_no_adherence_schedule_date_start(self):
+        self.assert_update(
+            (
+                datetime(2016, 1, 17),
+                (None, 'schedule1'),
+                []
+            ),
+            {
+            }
+        )
+
+    def test_no_adherence_cases(self):
+        self.assert_update(
+            (
+                datetime(2016, 1, 20),
+                (datetime(2016, 1, 10), 'schedule1'),
+                []
+            ),
+            {   # change
+                'aggregated_score_date_calculated': date(2016, 1, 20),
+                'expected_doses_taken': (10.0 / 7) * int(self.fixture_data['schedule1']),
+                'aggregated_score_count_taken': 0,
+                # 1 day before should be schedule-date-start
+                'adherence_latest_date_recorded': date(2016, 1, 9),
+                'adherence_total_doses_taken': 0
+            }
+        )
+
+    def test_adherence_date_less_than_purge_date(self):
+        self.assert_update(
+            (
+                datetime(2016, 1, 20),
+                (datetime(2016, 1, 10), 'schedule1'),
+                [(datetime(2016, 1, 15), DOSE_TAKEN_INDICATORS[0])]
+            ),
+            {
+                'aggregated_score_date_calculated': date(2016, 1, 15),
+                'expected_doses_taken': (5.0 / 7) * int(self.fixture_data['schedule1']),
+                'aggregated_score_count_taken': 1,
+                'adherence_latest_date_recorded': date(2016, 1, 15),
+                'adherence_total_doses_taken': 1
+            }
+        )
+
+    def test_adherence_date_greater_than_purge_date(self):
```

Comment:
Great catch on this test case! Looks right to me, and is an important one to identify